### PR TITLE
[PORT] New Heretic Path: Cosmic

### PR DESCRIPTION
**Pull request by cowbot92**
Full credit to all code and sprites go to the original PR: https://github.com/tgstation/tgstation/pull/73915

# Document the changes in your pull request

Adds several new abilities + a new heretic path- the cosmic path!

Base Lore - Eternal Gate -  Allows you to transmute a sheet of plasma and a knife into an Cosmic Blade. Additionally your grasp will now cause a cosmic ring on your targets, which are affected by your abilities.

T1 - Cosmic Runes - Grants you Cosmic Runes, a spell that creates two runes linked with each other for easy teleportation.  Only the entity activating the rune will get transported, and it can be used by anyone without a star mark. However, people with a star mark will get transported along with another person using the rune."

Grasp Mark - Mark of Cosmos - Your Mansus Grasp now applies the Mark of Cosmos. The mark is triggered from an attack with your Cosmic Blade. When triggered, the victim is returned to the location where the mark was originally applied to them. They will then be paralyzed for 2 seconds

T2 - Star Blast - Fires a projectile that moves very slowly and creates cosmic fields on impact. Anyone hit by the projectile will receive burn damage, a knockdown, and give people in a three tile range a star mark.

Blade Upgrade - Stellar Sublimation- Your blade now deals damage to people's cells through cosmic radiation.

T3- Cosmic Expansion - Grants you Cosmic Expansion, a spell that creates a 3x3 area of cosmic fields around you. Nearby beings will also receive a star mark.

Ascension Rite - Creator's Gift - The ascension ritual of the Path of Cosmos. Bring 3 corpses to a transmutation rune to complete the ritual. When completed, you may transform into a Star Gazer. The Star Gazer has an aura that will heal you and damage opponents. Optionally, you may summon a Star Gazer Ally instead, and gain damage reduction, stun immunity, and space immunity."

https://github.com/yogstation13/Yogstation/assets/75333826/8eddd413-72b2-4ca4-8aba-ed4dcb90b685

# Why is this good for the game?

More paths! This path has a lot of good options for locking down areas- which heretics don't have except for a few void spells. The teleport runes in theory should also be interesting. Basically- more tools to bring to the show!

# Testing

Refer to the video for all tested work, but yes I made sure everything works as it should on the tin.

# Spiriting

Star_Gazer
Cosmic_Blade
Cosmic_Blade_Showcase
cosmic_carpet
star_ball
cosmic_diamond
space_explosion
cosmic_ring
cosmic_mark
cosmic_cloud
image

# Wiki Documentation

I will update the wiki once I finish with the remaining porting.

# Changelog

:cl:  
rscadd: Adds new Heretic paths
soundadd: Adds new heretic sounds
imageadd: added new sprites, icons, and actions
/:cl:

### CODE REVIEW

**Comment by Yarinoi** [comment]
day by day we get closer to cowbot coding knock path

File: code/modules/antagonists/eldritch_cult/knowledge/cosmo_lore.dm [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+/datum/eldritch_knowledge/base_cosmic/on_lose(mob/user)
+	UnregisterSignal(user, COMSIG_HERETIC_MANSUS_GRASP_ATTACK)
+
+/// Aplies the effect of the mansus grasp when it hits a target.
+// /datum/eldritch_knowledge/cosmic_grasp/proc/on_mansus_grasp(mob/living/source, mob/living/target)
+// 	SIGNAL_HANDLER
+
+// 	to_chat(target, span_danger("A cosmic ring appeared above your head!"))
+// 	target.apply_status_effect(/datum/status_effect/star_mark, source)
+// 	new /obj/effect/forcefield/cosmic_field(get_turf(source))
```
Comment (Moltijoe) [inline code/modules/antagonists/eldritch_cult/knowledge/cosmo_lore.dm]: why is this here, also, it's supposed to place a tile under the caster? not the target?

File: code/modules/antagonists/eldritch_cult/transmutations/cosmo_transmutations.dm [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	result_atoms = list(/obj/item/melee/sickly_blade/cosmic)
+	required_shit_list = "A pierce of plasma and a knife."
+
+/datum/eldritch_transmutation/final/cosmic_final
+	name = "Creators's Gift"
+	required_atoms = list(/mob/living/carbon/human)
+	required_shit_list = "Three dead bodies."
+
+/datum/eldritch_transmutation/final/cosmic_final/on_finished_recipe(mob/living/user, list/atoms, loc)
+	var/alert_ = tgui_alert(user, "Do you want to ascend as a Star Gazer, or instead empower yourself and summon one as an ally?", "...", list("Yes","No"))
```
Comment (Moltijoe) [inline code/modules/antagonists/eldritch_cult/transmutations/cosmo_transmutations.dm]: do you want to do this or that?
yes/no

File: code/modules/antagonists/eldritch_cult/transmutations/cosmo_transmutations.dm [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+			var/datum/antagonist/heretic_monster/monster = summoned.mind.has_antag_datum(/datum/antagonist/heretic_monster)
+			var/datum/antagonist/heretic/master = user.mind.has_antag_datum(/datum/antagonist/heretic)
+			monster.set_owner(master)
+			master.ascended = TRUE
+		if("Yes")
+			var/mob/living/summoned = new /mob/living/simple_animal/hostile/eldritch/star_gazer(loc,TRUE,10)
+			summoned.ghostize(0)
+			user.SetImmobilized(0)
+			for(var/datum/action/cooldown/spell/spells in user.actions)
+				if(istype(spells, /datum/action/cooldown/spell/jaunt/ethereal_jaunt/ash)) //vitally important since ashen passage breaks the shit out of armsy
```
Comment (Moltijoe) [inline code/modules/antagonists/eldritch_cult/transmutations/cosmo_transmutations.dm]: armsy? i assume this code was copied from flesh, jaunts shouldn't be an issue with a single large simplemob

**Review by Moltijoe — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/eldritch_cult/knowledge/cosmo_lore.dm (hunk 1 above)
Comment (cowbot92) [reply code/modules/antagonists/eldritch_cult/knowledge/cosmo_lore.dm]: sorry that is the original code from TG that I had to adjust for our grasp system- and yes it spawns under the caster to better control it

File: code/modules/antagonists/eldritch_cult/transmutations/cosmo_transmutations.dm (hunk 2 above)
Comment (cowbot92) [reply code/modules/antagonists/eldritch_cult/transmutations/cosmo_transmutations.dm]: ok fair

File: code/modules/antagonists/eldritch_cult/transmutations/cosmo_transmutations.dm (hunk 3 above)
Comment (cowbot92) [reply code/modules/antagonists/eldritch_cult/transmutations/cosmo_transmutations.dm]: all the spells get removed once you transform anyways, but this is fair

Comment (cowbot92) [reply code/modules/antagonists/eldritch_cult/transmutations/cosmo_transmutations.dm]: I'll adjust the comment 'cause I dont really like the big mobs being able to use it anyways though

**Review by Moltijoe — APPROVED** [review]
lots of code, but it all seems pretty simple
there's bound to be at least 1 bugfix pr after this i'm sure

**Comment by ManofSan** [comment]
domain expansion

**Comment by Xoxeyos** [comment]
pretty epic, excited to see this ing

**Comment by cowbot92** [comment]
fixed merge conflicts

**Review by cuackles — APPROVED** [review]

**Comment by bruhlookatthisdood** [comment]
question, how much health does the star gazer thing have
and can I maxcap it to save the station

**Review by ToasterBiome — APPROVED** [review]
